src/main/java/org/datadog/jmxfetch/AttachApiConnection.java

```diff
@@ -49,7 +52,12 @@ private String getJMXUrlForProcessRegex(String processRegex) throws com.sun.tool
                 return connectorAddress;
             }
         }
-        throw new IOException("Cannot find JVM matching regex: " + processRegex);
+
+         List<String> jvms = new ArrayList<String>();
+         for (com.sun.tools.attach.VirtualMachineDescriptor vmd : com.sun.tools.attach.VirtualMachine.list()) {
+             jvms.add( vmd.displayName() );
+         }
+        throw new IOException("Cannot find JVM matching regex: '" + processRegex + "'; available JVMs (for this user account): " + jvms );
```

Comment:
Looks like this test requires the same update https://github.com/DataDog/jmxfetch/blob/master/src/test/java/org/datadog/jmxfetch/TestServiceChecks.java#L116